src/components/ObsDetails/ObsDetails.js

```diff
@@ -70,31 +71,84 @@ const ObsDetails = ( ): Node => {
   const bottomSheetModalRef = useRef( null );
   const [addingComment, setAddingComment] = useState( false );
   const [snapPoint, setSnapPoint] = useState( 100 );
-  const apiToken = useApiToken( );
-  const [localObservation, setLocalObservation] = useState( null );
+  const realm = useRealm( );
+  const localObservation = realm?.objectForPrimaryKey( "Observation", uuid );
```

Comment:
This fetches the local observation on every render. Why do we need to do that?

Reply:
We don't necessarily need to, but this fixes a bug where `ObsDetails` can't load because of the way state is set for `localObservation`. We're currently getting a `Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.` error and a white screen on observations made locally.

Reply:
Was that coming from https://github.com/inaturalist/iNaturalistReactNative/pull/206/files#diff-f3017450aeb5c7dc7b74f3e729e38fce3460974a5a5344677a088d935d568e82L87-L89? I can't replicate that bug if I add something like that back, but wouldn't the solution there be to figure how why the items in the dependency array keep changing?

Reply:
I guess this doesn't need to block, but this still seems like a problem to me, or rather a solution to a problem I can't replicate.